integration/mediation-tests/tests-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/GreenMailUtil.java

```diff
@@ -0,0 +1,308 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.esb.integration.common.utils;
+
+import com.icegreen.greenmail.user.GreenMailUser;
+import com.icegreen.greenmail.user.UserException;
+import com.icegreen.greenmail.util.GreenMail;
+import com.icegreen.greenmail.util.ServerSetup;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.esb.integration.common.utils.exception.ESBMailTransportIntegrationTestException;
+
+import java.util.Date;
+import java.util.Map;
+import java.util.Properties;
+import javax.mail.Flags;
+import javax.mail.Folder;
+import javax.mail.Message;
+import javax.mail.MessagingException;
+import javax.mail.Session;
+import javax.mail.Store;
+import javax.mail.URLName;
+import javax.mail.internet.InternetAddress;
+import javax.mail.internet.MimeMessage;
+import javax.mail.search.AndTerm;
+import javax.mail.search.BodyTerm;
+import javax.mail.search.FlagTerm;
+import javax.mail.search.SearchTerm;
+import javax.mail.search.SubjectTerm;
+
+/**
+ * Util class for GreenMail Server
+ */
+public class GreenMailUtil {
+    protected static Log log = LogFactory.getLog(GreenMailUtil.class);
+    private static GreenMail greenMail;
+    private static GreenMailUser greenMailUser;
+    private static String EMAIL_INBOX = "INBOX";
+    private static String EMAIL_TRASH = "Trash";
+    private static int WAIT_TIME_MS = 180 * 1000;
+
+    /**
+     * Start the server and add the user
+     */
+    public static void startServer() {
+        ServerSetup popSetup = new ServerSetup(3110, "localhost", "pop3");
+        ServerSetup imapSetup = new ServerSetup(3143, "localhost", "imap");
+        ServerSetup[] setups = new ServerSetup[2];
+        setups[0] = popSetup;
+        setups[1] = imapSetup;
+        greenMail = new GreenMail(setups);
+        greenMail.start();
+        greenMailUser = greenMail.setUser("wso2@localhost", "wso2", "wso2");
+        log.info("GreenMail Server started and user added!");
+    }
+
+    /**
+     * Stop server
+     */
+    public static void stopServer() {
+        greenMail.stop();
+        log.info("GreenMail Server stopped!");
+    }
+
+    /**
+     * Sending email to the user account in the server
+     * @param subject Email subject
+     * @throws MessagingException
+     * @throws UserException
+     * @throws InterruptedException
+     */
+    public static void sendMail(String subject) throws MessagingException, UserException, InterruptedException {
+        MimeMessage message = new MimeMessage((Session) null);
+        message.setFrom(new InternetAddress("some@mail.com"));
+        message.addRecipient(Message.RecipientType.TO, new InternetAddress("wso2@localhost"));
+        message.setSubject(subject);
+        message.setText("Content :" + subject);
+        message.setSentDate(new Date());
+
+        greenMailUser.deliver(message);
+        Thread.sleep(10000);
```

Comment:
Shall we use some other mechanism to verify the email delivery instead of thread sleep. Basically adding a new utils method to check whether email delivered or not in a busy loop?

Reply:
done